### PR TITLE
Throw TypeError on delete with null/undefined base

### LINKED ISSUE #544: delete on null/undefined base should throw TypeError for both string and symbol computed keys
## Engine bug: `delete null[x]` and `delete undefined[x]` silently return `true` instead of throwing TypeError

Per ES2026 §13.5.1.2 (delete operator) the property reference is evaluated first; accessing a property on `null`/`undefined` throws `TypeError` before delete semantics apply. V8/Node:

```js
delete null["x"];        // TypeError: Cannot convert undefined or null to object
delete undefined["x"];   // TypeError
delete null[Symbol()];   // TypeError
```

Goccia returns `true` for all of these, both in interpreter and bytecode mode.

### Reproduce

```bash
./build.pas loader
cat > /tmp/probe.js <<'EOJS'
const sym = Symbol();
console.log("delete null[\"x\"]:", delete null["x"]);          // true (BUG, should TypeError)
console.log("delete undefined[\"x\"]:", delete undefined["x"]); // true (BUG)
console.log("delete null[sym]:", delete null[sym]);             // true (BUG)
console.log("delete undefined[sym]:", delete undefined[sym]);   // true (BUG)
EOJS
./build/GocciaScriptLoader /tmp/probe.js
./build/GocciaScriptLoader --mode=bytecode /tmp/probe.js
```

### Spec context

`delete obj[expr]`:
1. Let baseRef = the base of the member expression (the receiver).
2. If `baseRef` is `null` or `undefined`, throw `TypeError` (RequireObjectCoercible at the property-access point).
3. Otherwise apply `[[Delete]]`.

The throw is at the property-access stage, before delete semantics ever run. The current Goccia behaviour skips the receiver-coercibility check for the no-op branch.

### Likely fix area

`source/units/Goccia.Evaluator.pas` `EvaluateDelete` — add a shared nullish-base guard before the symbol/string key split:

```pascal
if (ObjValue is TGocciaNullLiteralValue) or
   (ObjValue is TGocciaUndefinedLiteralValue) then
  ThrowTypeError(Format(SErrorCannotReadPropertiesOf, [<key-display>, <obj-display>]),
    SSuggestCheckNullBeforeAccess);
```

Plus the equivalent in the bytecode VM `OP_DEL_INDEX`. Currently the receiver-not-Object branch in the interpreter falls through to `Result := TGocciaBooleanLiteralValue.TrueValue` without coercibility check; bytecode mirrors that.

### Tests

Add to `tests/language/expressions/delete/delete-operator.js`:

```js
test("delete null[\"x\"] throws TypeError", () => {
  expect(() => { delete null["x"]; }).toThrow(TypeError);
});
test("delete undefined[\"x\"] throws TypeError", () => {
  expect(() => { delete undefined["x"]; }).toThrow(TypeError);
});
test("delete null[sym] throws TypeError", () => {
  const sym = Symbol();
  expect(() => { delete null[sym]; }).toThrow(TypeError);
});
test("delete undefined[sym] throws TypeError", () => {
  const sym = Symbol();
  expect(() => { delete undefined[sym]; }).toThrow(TypeError);
});
```

### Scope notes

- Pre-existing in both modes; CodeRabbit raised on the `EvaluateDelete` symbol-key migration in #527, deferred there because it is unrelated to the `ToStringLiteral` / `ToPropertyKey` migration this PR is scoped to.
- Behaviour change: any code currently relying on `delete null[...]` silently succeeding will now throw. Per spec this is the correct behaviour and is what V8/Node already do.

### Surfaced by

#527 review feedback ([discussion](https://github.com/frostney/GocciaScript/pull/527#discussion_r3191448696)).

### Related

- #522 — `String(obj)` ES2026 §7.1.17 ToString.
- The symbol-keyed deletion path was added in #527 — this issue is the spec-correctness gate that should cover both the symbol and string paths uniformly.

### PR DESCRIPTION
**Pull request by frostney**
## Summary

- Add RequireObjectCoercible guard in `EvaluateDelete` (interpreter) and `OP_DELETE_PROP_CONST`/`OP_DEL_INDEX` (bytecode VM) so that `delete null[key]`, `delete undefined[key]`, and dot-notation variants throw TypeError instead of silently returning `true`
- Covers both string and symbol computed keys, matching ES2026 §13.5.1.2
- Add `KeyDisplaySafe` helper in the VM to extract a key display string without invoking ToPrimitive, preserving spec evaluation order in the error path
- Add 6 new tests covering all combinations (null/undefined × string/symbol/dot)

Closes #544

## Testing

- [x] Verified no regressions and confirmed the new feature or bugfix in end-to-end JavaScript/TypeScript tests
- [ ] Updated documentation
- [ ] Optional: Verified no regressions and confirmed the new feature or bugfix in native Pascal tests (if AST, scope, evaluator, or value types changed)
- [ ] Optional: Verified no benchmark regressions or confirmed benchmark coverage for the change

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: source/units/Goccia.VM.pas [hunk 1]
```diff
@@ -8231,11 +8239,15 @@       // ES2022 §15.7.14: execute static block closure with this = class
...
-        // `delete arr[sym]` doesn't silently no-op via the array fall-through.
-        if (FRegisters[B].Kind = grkObject) and
+        if FRegisters[B].Kind = grkNull then
+          ThrowTypeError(Format(SErrorCannotReadPropertiesOfNull,
+            [KeyToPropertyNameRegister(FRegisters[C])]),
+            SSuggestCheckNullBeforeAccess)
+        else if FRegisters[B].Kind = grkUndefined then
+          ThrowTypeError(Format(SErrorCannotReadPropertiesOfUndefined,
+            [KeyToPropertyNameRegister(FRegisters[C])]),
+            SSuggestCheckNullBeforeAccess)
```
[the remark this answers is not in the document]
Comment (frostney) [reply source/units/Goccia.VM.pas:8275]: Fixed in 4d0cc1fd. Added `KeyDisplaySafe` helper that extracts a display string from the key register without invoking ToPrimitive — returns the literal value for strings, symbols, and primitives, falls back to `<computed>` for object-kind registers that would require coercion.

**Comment by frostney** [comment]
@coderabbitai review
